GrayBMP.cs

```diff
@@ -105,6 +106,22 @@ class GrayBMP {
       End ();
    }
 
+   void DrawHLine (int x1, int x2, int y, int gray) {
+      Begin ();
+      if (x2 < x1) (x1, x2) = (x2, x1);
+      Check (x1, y); Check (x2, y);
+      Dirty (x1, y, x2, y); byte bGray = (byte)gray;
+      unsafe {

```

Comment:
Why not write this as a for loop? You know exactly how many pixels are going to be plotted.